app/src/main/java/com/expidev/gcmapp/support/v4/fragment/SettingsFragment.java

```diff
@@ -169,6 +187,16 @@ private void updateMccPrefSummary() {
         }
     }
 
+    private class MinistryChangeListener implements Preference.OnPreferenceChangeListener {
+        @Override
+        public boolean onPreferenceChange(Preference preference, Object newValue) {
+            // update summary as necessary
+            updateMinistryPrefSummary();
```

Comment:
value is persisted before the change method is triggered. So, this should be good